test/tests/parallel/set1/spam_virus/disabled_clamav_spamassassin.bats

```diff
@@ -12,12 +12,14 @@ function setup_file() {
     --env ENABLE_CLAMAV=0
     --env ENABLE_SPAMASSASSIN=0
     --env AMAVIS_LOGLEVEL=2
+    --env PERMIT_DOCKER=container
```

Comment:
Was this test not working properly before? Was the `PERMIT_DOCKER` needed? Or is that related to change to `_send_email` with `swaks`? 🤔

Reply:
We now get

```txt
450 4.1.8 <user@external.tld>: Sender address rejected: Domain not found
```

which is what we are seeing with Postscreen as well. We need to agree on which solution we want to settle for in this PR. It seems to me the answer to your question is actually "yes", which worries me. The transactions should succeed...

Reply:
> It seems to me the answer to your question is actually "yes", which worries me.
> The transaction needs to succeed in order for the test to work...

Does it?

I can't say for sure with the third test-case there, but the other two are part of startup, which isn't dependent upon mail being sent. Oddly that third test case is looking for failure to find the log entry about not being able to connect to clamav, rather than success, or either outcome... so it's probably not a good test-case in the first place 🤔 

---

In this case `PERMIT_DOCKER=container` is only for granting the success to deliver the mail within the same container. An action that may have no value to the test. depending on requirements for actually triggering a failure on the third-test case (_if the SA + ClamAV services were enabled_).

If it's not dependent, we don't need to send an email at all for this.

---

For Postscreen test, we have a separate container, which would need `PERMIT_DOCKER` to grant subnet trust to workaround I guess. We already have containers using `PERMIT_DOCKER` this way elsewhere, it's acceptable until we can leverage DNS.

It's easy enough to use `--add-host external.tld:127.0.0.1` in `setup.bash` for each container to have `/etc/hosts` append that entry, but when we need to handle DNS records for SPF that won't be enough.

BATS does have support for a pre-setup/post-teardown to run at start and end of the entire run.
- That would allow setting up CoreDNS and modifying `setup.bash` to have containers use `--dns` to point to it.
- It can only support something like `external.tld` which can have a static IP, whereas DNS for `--to`/`--server` of `swaks` would use the same domains across containers, and we can't easily map the IP from CoreDNS to each of those.
- So this is technically viable for `external.tld`, but you'd also need to route all mail through that shared container across tests, which may not be ideal.

Alternatively, we could create separate CoreDNS containers at the test file scope, but we'd still need to get the IP per container 😅 

We can probably query that with `docker inspect` and `jq` (_requires starting a container, then updating CoreDNS_), or by setting a static IP (_not ideal to manage per container_). It's a bit tricky? Mostly because we're running tests concurrently which complicates managing those concerns.

A hybrid may work with single CoreDNS and `--add-host`, FQDN is resolved to IP via `/etc/hosts`, and other DNS records like `TXT` for SPF via CoreDNS? I haven't tried that combination yet 🤔 

Anyway for now we can just rely on `PERMIT_DOCKER` as we currently do.